src/core/tests/queryCache.test.js

```diff
@@ -224,4 +228,15 @@ describe('queryCache', () => {
     expect(newQuery.state.markedForGarbageCollection).toBe(false)
     expect(newQuery.state.data).toBe('data')
   })
+
+  test('makeQueryCache merges defaultConfig so providing a queryFn does not overwrite the default queryKeySerializerFn', async () => {
+    const queryFn = () => 'data'
+    const queryCache = makeQueryCache({
+      defaultConfig: { queries: { queryFn } },
+    })
+
+    expect(() => queryCache.buildQuery('test')).not.toThrow(
+      'config.queryKeySerializerFn is not a function'
+    )
```

Comment:
Open to suggestions on a better way to validate this -- I started by writing this test expecting the exception because that is the current behavior, switched to `.not.toThrow` and applied my code changes.

Reply:
I'm not sure either. But we can make it better in the future if we need to. If tests pass and you think this is good enough, then let me know and I'll merge :)

Reply:
If you're happy with it, the tests pass and it resolves the issue I was having :)
I had also considered exposing `configRef` on the `queryCache` but wasn't sure if increasing the public api to support testing was a good idea 🤔

Reply:
I'd rather not expose more api lol. I'm doing too much of that as it is.